### ISSUE TITLE
Inconsistency due to `rdfs:domain` of `rml:logicalTarget` and its use on `rml:LanguageMap`

### ISSUE THREAD

**Issue by pmaria** [kg-construct/rml-io#61]
The `rdfs:domain` of `rml:logicalTarget` is defined to be `rml:TermMap`, yet it is also used on `rml:LanguageMap` which is not a `rml:TermMap`. 

Options:
* drop the domain for `rml:logicalTarget`
* make the domain a union of `rml:TermMap` and `rml:LanguageMap`

**Comment by andimou** [comment]
couldn't we have another option to have the domain as `rml:ExpressionMap`?

**Comment by pmaria** [comment]
> couldn't we have another option to have the domain as `rml:ExpressionMap`?

That would be too broad, since there are expression maps that do not directly cause the generation of a term (part), like `rml:ChildMap` and `rml:ParentMap`.

**Comment by anaigmo** [comment]
I agree with @pmaria in this, to have as domain the union of both instead of `rml:ExpressionMap` is more accurate. I'll add this and the other pending changes to the IO ontology this week

**Comment by andimou** [comment]
right and what about the DatatypeMap? I.e. why only the LanguageMap and not the DatatypeMap?

**Comment by pmaria** [comment]
Yes, good question. I think we can see DatatypeMap as a TermMap in the sense that it always generates an IRI. It definitely also generates a part of a literal term. I don't believe we currently classed it as a TermMap. So maybe we should raise an issue for this.

**Comment by andimou** [comment]
that's an interesting observation, 👍 on raising an issue for this.

I didn't exactly mean this though 😅 I meant a union of `rml:TermMap`, `rml:LanguageMap` and `rml:DatatypeMap`.
If one can define a larget in the `rml:LanguageMap`, then it should be the same for a `rml:DatatypeMap` no?

**Comment by pmaria** [comment]
I agree, we should also be able to define a target on a `rml:DatatypeMap` if we are allowing a target on a `rml:LanguageMap`

**Comment by dachafra** [comment]
I think this is solved but in a PR in the `rml-core`, @DylanVanAssche could you confirm this? Do we need to update anything in this repo?
